### PR TITLE
CPM-634: Replace calls to fromIdentifier(s) with fromUuid(s) for connector products

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

Remove methods in GetConnectorProducts that use product identifiers, and replace all calls with methods using Uuids instead

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by cemorin — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProducts.php [hunk 1]
```diff
@@ -145,7 +131,7 @@ public function fromProductUuids(
 
         $products = [];
         foreach ($productUuids as $productUuid) {
-            if (!isset($rows[$productUuid->toString()])) {
+            if (!isset($rows[$productUuid->toString()]) || !isset($rows[$productUuid->toString()]['uuid'])) {
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProducts.php:134]: Check why we need this

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProductsWithOptions.php [hunk 2]
```diff
@@ -10,6 +10,7 @@
 use Akeneo\Pim\Enrichment\Component\Product\Query\ProductQueryBuilderInterface;
 use Akeneo\Pim\Enrichment\Component\Product\Value\OptionsValue;
 use Akeneo\Pim\Enrichment\Component\Product\Value\OptionValue;
+use Akeneo\Pim\Enrichment\Product\Infrastructure\Query\GetProductUuidsFromProductIdentifiers;
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProductsWithOptions.php:13]: Remove use

File: src/Akeneo/Pim/Enrichment/Product/back/Infrastructure/Query/GetProductUuidsFromProductIdentifiers.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+use Doctrine\DBAL\Connection;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class GetProductUuidsFromProductIdentifiers
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Product/back/Infrastructure/Query/GetProductUuidsFromProductIdentifiers.php:15]: Remove and replace calls with SqlFindProductUuids

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Query/GetProductUuidsFromProductIdentifiersIntegration.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Test\Pim\Enrichment\Product\Integration\Query;
+
+use Akeneo\Pim\Enrichment\Product\Infrastructure\Query\GetProductUuidsFromProductIdentifiers;
+use Akeneo\Test\Pim\Enrichment\Product\Integration\EnrichmentProductTestCase;
+
+final class GetProductUuidsFromProductIdentifiersIntegration extends EnrichmentProductTestCase
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Query/GetProductUuidsFromProductIdentifiersIntegration.php:10]: To delete if we remove query

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php [hunk 5]
```diff
@@ -339,7 +211,7 @@ public function getAction(Request $request, string $code): JsonResponse
             if ($request->query->getAlpha('with_completenesses', 'false') === 'true') {
                 $product = $this->getProductsWithCompletenesses->fromConnectorProduct($product);
             }
-        } catch (ObjectNotFoundException $e) {
+        } catch (ObjectNotFoundException | \InvalidArgumentException $e) {
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php:214]: Where this comes from ?

**Review by pierallard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProducts.php (hunk 1 above)
Comment (SamirBoulil) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Connector/SqlGetConnectorProducts.php:134]: Yes I don't think you need it isset($array['1']['2']); returns false even if neither of '1' or '2' key is set

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php (hunk 5 above)
Comment (cemorin) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php]: I didn't find where it came from and the error is well handled with the `ObjectNotFoundException` so I removed the `InvalidArgumentException`

**Pull request merged**